includes/handler/class-announce.php

```diff
@@ -40,6 +42,13 @@ public static function handle_announce( $announcement, $user_ids, $activity = nu
 			return;
 		}
 
+		// Ignore announces from the blog actor.
+		$actor_id = object_to_uri( $announcement['actor'] );
+		$user_id  = Actors::get_id_by_resource( $actor_id );
+		if ( is_same_domain( $actor_id ) && ! \is_wp_error( $user_id ) && Actors::BLOG_USER_ID === $user_id ) {
```

Comment:
```suggestion
		if ( Actors::BLOG_USER_ID === Actors::get_id_by_resource( $announcement['actor'] ) ) {
```
Is all of this other stuff necessary? Tests still pass without it

Reply:
It is not necessary, but I thought it might speed up checks if we do not have the lookup when the domain is no local one.